src/main/java/io/cryostat/recordings/LongRunningRequestGenerator.java

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright The Cryostat Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.cryostat.recordings;
+
+import java.time.Duration;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+
+import io.cryostat.ConfigProperties;
+import io.cryostat.reports.ReportsService;
+import io.cryostat.ws.MessagingServer;
+import io.cryostat.ws.Notification;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import io.quarkus.vertx.ConsumeEvent;
+import io.vertx.mutiny.core.eventbus.EventBus;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import org.apache.commons.lang3.tuple.Pair;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@ApplicationScoped
+public class LongRunningRequestGenerator {
+
+    public static final String ARCHIVE_ADDRESS =
+            "io.cryostat.recordings.ArchiveRequestGenerator.ArchiveRequest";
+    public static final String GRAFANA_ARCHIVE_ADDRESS =
+            "io.cryostat.recordings.ArchiveRequestGenerator.GrafanaArchiveUploadRequest";
+    public static final String GRAFANA_ACTIVE_ADDRESS =
+            "io.cryostat.recordings.ArchiveRequestGenerator.GrafanaActiveUploadRequest";
+    public static final String ARCHIVE_REPORT_ADDRESS =
+            "io.cryostat.recordings.ArchiveRequestGenerator.ArchiveReportRequest";
+    public static final String ACTIVE_REPORT_ADDRESS =
+            "io.cryostat.recordings.ArchiveRequestGenerator.ActiveReportRequest";
+    private static final String ARCHIVE_RECORDING_SUCCESS = "ArchiveRecordingSuccess";
+    private static final String ARCHIVE_RECORDING_FAIL = "ArchiveRecordingFailed";
+    private static final String GRAFANA_UPLOAD_SUCCESS = "GrafanaUploadSuccess";
+    private static final String GRAFANA_UPLOAD_FAIL = "GrafanaUploadFailed";
+    private static final String REPORT_SUCCESS = "ReportSuccess";
+    private static final String REPORT_FAILURE = "ReportFailure";
+    private final ExecutorService executor;
+    private final Logger logger = LoggerFactory.getLogger(getClass());
```

Comment:
This can just be `@Inject Logger logger`, which might be a little more conventional looking.

Reply:
I think it needs to be a [`org.jboss.logging.Logger`](https://github.com/cryostatio/cryostat/blob/65e3787079d839626ae536954d1b06843ac6ba24/src/main/java/io/cryostat/recordings/ActiveRecordings.java#L56)